pkg/scheduler/framework/plugins/noderesources/fit.go

```diff
@@ -441,6 +448,39 @@ func (f *Fit) isSchedulableAfterNodeChange(logger klog.Logger, pod *v1.Pod, oldO
 	return fwk.Queue, nil
 }
 
+// isSchedulableAfterDeviceClassChange is invoked whenever a device class added or changed. It checks whether
+// that change could make a previously unschedulable pod schedulable.
+func (f *Fit) isSchedulableAfterDeviceClassEvent(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (fwk.QueueingHint, error) {
+	originalClass, modifiedClass, err := schedutil.As[*resourceapi.DeviceClass](oldObj, newObj)
+	if err != nil {
+		return fwk.Queue, err
+	}
+	if originalClass != nil && modifiedClass != nil && originalClass.Spec.ExtendedResourceName == modifiedClass.Spec.ExtendedResourceName {
+		return fwk.QueueSkip, nil
+	}
+	if originalClass != nil && modifiedClass != nil && originalClass.Spec.ExtendedResourceName != nil && modifiedClass.Spec.ExtendedResourceName != nil && *originalClass.Spec.ExtendedResourceName == *modifiedClass.Spec.ExtendedResourceName {
```

Comment:
This check and the one above look very similar. The change below can cover both:
```suggestion
	if originalClass != nil originalClass.Spec.ExtendedResourceName != nil && ptr.Deref(originalClass.Spec.ExtendedResourceName, "") == ptr.Deref(modifiedClass.Spec.ExtendedResourceName, "") {
```